### PR TITLE
Allow for non jll dependencies

### LINKED ISSUE #1132: Fictional UUID created for non JLL package
In a local version of https://github.com/JuliaPackaging/Yggdrasil/pull/4073 I am experimenting with adding a non `jll` dependency.
Thanks to @giordano who pointed me to the right place. We should look up the UUID instead of synthesizing a new one.

https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/084d75b38aa9d0a0240d328d9ff3adbc9c7d9068/src/AutoBuild.jl#L1442

cc: @staticfloat

### PR DESCRIPTION
**Pull request by vchuravy**
Fixes #1132

### CODE REVIEW

File: src/AutoBuild.jl [hunk 1]
```diff
@@ -1425,6 +1425,29 @@ const uuid_package = UUID("cfb74b52-ec16-5bb7-a574-95d9e393895e")
...
+        return pkg.uuid
+    end
+    depname = getname(pkg)
+    @static if VERSION >= v"1.7"
+        uuids = Pkg.Types.registered_uuids(ctx.registries, depname)
+    else
+        uuids = Pkg.Types.registered_uuids(ctx, depname)
+    end
+    if isempty(uuids)
+        return nothing
```
Comment (giordano) [inline src/AutoBuild.jl:1440]: I was thinking of using `jll_uuid` here, but maybe it's better to have this function returning the real UUID, if any, only.

**Review by giordano — APPROVED** [review]
Looks good to me, as long long as tests pass.  Thanks!

**Comment by giordano** [comment]
Yay, more internal and undocumented Pkg functions, to make future upgrades more fun 🥳

File: src/AutoBuild.jl (hunk 1 above)
Comment (vchuravy) [reply src/AutoBuild.jl:1440]: Yeah the first version did that, but then I was like: Maybe I want to use that function elsewhere xD

**Pull request merged**